extensions/amp-carousel/0.1/base-carousel.js

```diff
@@ -25,17 +25,66 @@ export class BaseCarousel extends AMP.BaseElement {
     /** @private {!Element} */
     this.nextButton_;
 
-    this.buildCarousel();
-    this.buildButtons();
-    this.setupGestures();
-    this.setControlsState();
+    /** @private {?number} */
+    this.autoplayTimeoutId_ = null;
```

Comment:
Yes, autoplay is somewhat logical. But looping is definitely not. Generally, I'd like to move toward isolating them more and more from each and not make them closer, since I think the original combination was a mistake.